tests/tests/test_state_scripts.py

```diff
@@ -1170,17 +1140,26 @@ def verify_script_log_correct(self, test_set, log):
                 if (log_pos < len(log)
                     and log[log_pos - 1].startswith("Sync_")
                     and log[log_pos].startswith("Idle_")
-                    and not expected_order[expected_pos].startswith("Idle_")):
+                    and not expected_order[expected_pos % len(expected_order)].startswith("Idle_")):
                     # The Idle/Sync sequence is allowed to "wrap around" and start
                     # over, because it may take a few rounds of checking before the
                     # deployment is ready for the client.
-                    expected_pos = idle_pos
+                    expected_pos = 0
 
-                if (expected_pos < len(expected_order)
-                    and not expected_order[expected_pos - 1].startswith("Idle_")
-                    and expected_order[expected_pos].startswith("Idle_")):
-                    # New Idle sequence entered.
-                    idle_pos = expected_pos
+                if expected_pos == len(expected_order):
```

Comment:
Minor: Is it possible to move this if-block up, to right after the `expected_pos = expected_pos + 1`. Then you will not need the `expected_pos % len(expected_order)` wraparound guard, and it should be a bit easier to read.

Reply:
I made the change and run all state script tests locally :+1: